core/src/main/scala/org/apache/spark/sql/execution/CodegenSparkFallback.scala

```diff
@@ -87,6 +88,15 @@ case class CodegenSparkFallback(var child: SparkPlan) extends UnaryExecNode {
             }
           case None => throw t
         }
+    } finally {
+      // SNAP-1422
+      val cacheMetaDataForConnector =
+        java.lang.Boolean.getBoolean("SMART_CONNECTOR_CACHE_METADATA")
+      lazy val session = sqlContext.sparkSession.asInstanceOf[SnappySession]
+      if (!cacheMetaDataForConnector &&
+        ExternalStoreUtils.isSmartConnectorMode(session.sparkContext)) {
+        session.sessionCatalog.invalidateAll()
```

Comment:
Initial idea as was that but it does not work as the error is thrown even before plan is created. Refer to the 2nd last comment in SNAP-1422(https://jira.snappydata.io/browse/SNAP-1422?focusedCommentId=16779&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16779)